src/Raven.Server/Documents/TimeSeries/TimeSeriesPolicyRunner.cs

```diff
@@ -0,0 +1,683 @@
+        }
+
+        internal class RollUpState
+        {
+            public Slice Key;
+            public string DocId;
+            public string Name;
+            public long Etag;
+            public string Collection;
+            public DateTime NextRollUp;
+            public string RollUpPolicy;
+            public string ChangeVector;
+        }
+
+        static TimeSeriesPolicyRunner()
+        {
+            using (StorageEnvironment.GetStaticContext(out var ctx))
+            {
+                Slice.From(ctx, "TimeSeriesRollUps", ByteStringType.Immutable, out TimeSeriesRollUps);
+                Slice.From(ctx, "RollUpKey", ByteStringType.Immutable, out RollUpKey);
+                Slice.From(ctx, "NextRollUpIndex", ByteStringType.Immutable, out NextRollUpIndex);
+                Slice.From(ctx, RawTimeSeriesPolicy.PolicyString, ByteStringType.Immutable, out RawPolicySlice);
+            }
+
+            RollUpSchema = new TableSchema();
+            RollUpSchema.DefineKey(new TableSchema.SchemaIndexDef
+            {
+                StartIndex = (int)RollUpColumn.Key,
+                Count = 1, 
+                Name = RollUpKey
+            });
+
+            RollUpSchema.DefineIndex(new TableSchema.SchemaIndexDef // this isn't fixed-size since we expect to have duplicates
+            {
+                StartIndex = (int)RollUpColumn.NextRollUp, 
+                Count = 1,
+                Name = NextRollUpIndex
+            });
+        }
+
+        public unsafe void MarkForPolicy(DocumentsOperationContext context, Slice key, string collection, string name, long etag, DateTime baseline, string changeVector)
+        {
+            var config = Configuration.Collections[collection];
+
+            if (config.Disabled)
+                return;
+
+            var current = config.GetPolicyByTimeSeries(name);
+            if (current == null) // policy not found
+                // TODO: delete this time-series if not the raw one?
+                return;
+
+            var nextPolicy = config.GetNextPolicy(current);
+            if (nextPolicy == null)
+                return;
+            
+            if (nextPolicy == TimeSeriesPolicy.AfterAllPolices)
+                return; // this is the last policy
+
+            var integerPart = baseline.Ticks / nextPolicy.AggregationTime.Ticks;
+            var nextRollup = nextPolicy.AggregationTime.Ticks * (integerPart + 1);
+
+            // mark for rollup
+            RollUpSchema.Create(context.Transaction.InnerTransaction, TimeSeriesRollUps, 16);
+            var table = context.Transaction.InnerTransaction.OpenTable(RollUpSchema, TimeSeriesRollUps);
+            using (table.Allocate(out var tvb))
+            using (Slice.From(context.Allocator, key.Content.Ptr, key.Size - sizeof(long) - sizeof(byte), ByteStringType.Immutable, out var timeSeriesKey))
+            using (Slice.From(context.Allocator,collection, ByteStringType.Immutable,out var collectionSlice))
```

Comment:
So we have a bigger problem here, we keep it with the original casing
e.g. https://github.com/ravendb/ravendb/blob/23c596b3b0f84bca9f7c19ec02e64a89850189a2/src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs#L906
Also, why do we store the collection at all? for replication only? 

Anyway, why do we care about it here? we need only to ensure that when we read from the configuration we do it in case-insensitive matter.